shaderc-rs/src/lib.rs

```diff
@@ -470,16 +497,18 @@ fn safe_str_from_utf8(bytes: &[u8]) -> String {
 }
 
 impl Compiler {
-    /// Returns an compiler object that can be used to compile SPIR-V modules.
+    /// Returns a `Compiler` object that can be used to compile SPIR-V modules.
     ///
-    /// A return of `None` indicates that there was an error initializing
-    /// the underlying compiler.
-    pub fn new() -> Option<Compiler> {
+    /// A return of `Err` indicates that there was an error initializing
+    /// the underlying compiler (i.e., `shaderc_compiler_initialize()` returned null).
+    pub fn new() -> Result<Self> {
         let p = unsafe { scs::shaderc_compiler_initialize() };
         if p.is_null() {
-            None
+            Err(Error::InitializationError(
+                "Failed to create a shaderc compiler.".to_string(),
```

Comment:
Nit: Use lower case "failed to create.." and drop the `.` at the end to be consistent with errors elsewhere so to compose nicer with potential message prefix/suffix in user code.

Reply:
Thank you!